clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -275,29 +316,34 @@ public void testHeartbeatNotSentIfAnotherOneInFlight() {
         assertEquals(1, result.unsentRequests.size());
         NetworkClientDelegate.UnsentRequest inflightReq = result.unsentRequests.get(0);
 
+        // Receive response for the inflight after the interval expired. The next HB should be sent
+        // on the next poll waiting only for the minimal backoff.
+        inflightReq.handler().onComplete(createHeartbeatResponse(inflightReq, Errors.NONE));
+        time.sleep(DEFAULT_RETRY_BACKOFF_MS);
+        when(heartbeatRequestState.canSendRequest(anyLong())).thenReturn(true);
```

Comment:
I wouldn't expect us to need to mock this canSend on the requestState that is a spy (the real method should kick in). But I guess you had to add it for some reason that was not generating the HB? I wonder if what we're missing is a time.sleep(HB_Interval) right before completing the response on ln 320? to ensure the interval has expired, the get a response, then check the next HB is sent only waiting for the backoff.

Reply:
Looks like at some point I changed the order, switched it back and it is working good now